sdgym/sdgym_result_explorer/result_handler.py

```diff
@@ -32,6 +37,106 @@ def load_synthetic_data(self, file_path):
+            column_name = (
+                f'{date_str}'
+                f' - # datasets: {folder_infos[folder]["# datasets"]}'
+                f' - sdgym version: {folder_infos[folder]["sdgym_version"]}'
+            )
+            results = results.loc[results['Synthesizer'] != SYNTHESIZER_BASELINE]
+            column_data = results.groupby(['Synthesizer'])['Win'].sum()
+            columns.append((date_obj, column_name, column_data))
+
+        columns.sort(key=operator.itemgetter(0))
+        summarized_results = pd.DataFrame()
+        for _, column_name, column_data in reversed(columns):
+            summarized_results[column_name] = column_data
+
+        summarized_results = summarized_results.fillna('-')
+        return summarized_results
+
+    def _get_column_name_infos(self, folder_to_results):
+        folder_to_info = {}
+        for folder, results in folder_to_results.items():
+            yaml_files = self._get_results_files(folder, prefix='run_', suffix='.yaml')
+            if not yaml_files:
+                continue
+
+            run_id_info = self._load_yaml_file(folder, yaml_files[0])
+            num_datasets = results.loc[
+                results['Synthesizer'] == SYNTHESIZER_BASELINE, 'Dataset'
+            ].nunique()
+            folder_to_info[folder] = {
+                'date': run_id_info.get('starting_date')[:10],
+                'sdgym_version': run_id_info.get('sdgym_version'),
+                '# datasets': num_datasets,
+            }
+
+        return folder_to_info
+
+    def summarize(self, folder_name):
+        """Summarize the results in the specified folder."""
+        all_folder = [f for f in self.list() if f.startswith(folder_name[:-11])]
+        if folder_name not in all_folder:
+            raise ValueError(f'Folder "{folder_name}" does not exist in the results directory.')
+
+        date = pd.to_datetime(folder_name[-10:], format='%m_%d_%Y')
+        folder_to_results = {}
+        for folder in all_folder:
+            folder_date = pd.to_datetime(folder[-10:], format='%m_%d_%Y')
+            if folder_date > date:
+                continue
+
+            result_filenames = self._get_results_files(folder, prefix='results_', suffix='.csv')
+            if not result_filenames:
+                continue
+
+            results = self._get_results(folder, result_filenames)
+            if not results:
+                continue
+
+            aggregated_results = pd.concat(results, ignore_index=True)
+            aggregated_results = aggregated_results.drop_duplicates(
+                subset=['Dataset', 'Synthesizer']
+            )
+            self._compute_wins(aggregated_results)
+            folder_to_results[folder] = aggregated_results
+            folder_infos = self._get_column_name_infos(folder_to_results)
+
+        summarized_table = self._get_summarize_table(folder_to_results, folder_infos)
+
+        return summarized_table
```

Comment:
can we return the concatenated results as well? As in a dataframe with the sdgym results from all runs in the folder concatenated